packages/fetchai/skills/simple_oracle/behaviours.py

```diff
@@ -0,0 +1,232 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a simple Fetch oracle contract deployment behaviour."""
+
+from typing import Any, Dict, cast
+
+from aea.mail.base import EnvelopeContext
+from aea.skills.behaviours import TickerBehaviour
+
+from packages.fetchai.connections.ledger.base import CONNECTION_ID as LEDGER_API_ADDRESS
+from packages.fetchai.contracts.oracle.contract import PUBLIC_ID as CONTRACT_PUBLIC_ID
+from packages.fetchai.protocols.contract_api.message import ContractApiMessage
+from packages.fetchai.protocols.ledger_api.message import LedgerApiMessage
+from packages.fetchai.skills.simple_oracle.dialogues import (
+    ContractApiDialogue,
+    ContractApiDialogues,
+    LedgerApiDialogues,
+)
+from packages.fetchai.skills.simple_oracle.strategy import Strategy
+
+
+DEFAULT_UPDATE_INTERVAL = 5
+
+
+class FetchOracleDeployer(TickerBehaviour):
+    """This class implements a behaviour that deploys a Fetch oracle contract."""
+
+    def __init__(self, **kwargs):
+        """Initialise the behaviour."""
+        update_interval = kwargs.pop(
+            "update_interval", DEFAULT_UPDATE_INTERVAL
+        )  # type: int
+        super().__init__(tick_interval=update_interval, **kwargs)
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+
+        self.context.logger.info("Setting up Fetch oracle contract...")
+        strategy = cast(Strategy, self.context.strategy)
+
+        if not strategy.is_contract_deployed:
+            self._request_contract_deploy_transaction()
+        else:
+            self.context.logger.info("Fetch oracle contract address already added")
+
+        if strategy.is_oracle_role_granted:
+            self.context.logger.info("Oracle role already granted")
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+
+        strategy = cast(Strategy, self.context.strategy)
+
+        if not strategy.is_contract_deployed:
+            self.context.logger.info("Oracle contract not yet deployed")
+            return
+
+        if not strategy.is_oracle_role_granted:
+            self.context.logger.info("Oracle not yet created")
+            self._request_grant_role_transaction()
+            return
+
+        # Check for entropy value from fetch oracle skill
+        oracle_data = self.context.shared_state.get("oracle_data", None)
+        if oracle_data is None:
+            self.context.logger.info("No oracle value to publish")
+        else:
+            self.context.logger.info("Publishing oracle value")
+
+            # add expiration block
+            update_args = oracle_data.copy()
+            update_args["expiration_block"] = 1000000000000000
```

Comment:
should this be configurable maybe?

Reply:
Yes, good catch, I've meaning to add this and also the address below to the config.